### PR TITLE
adds npmignore as it is not bundling when published to npm

### PR DESCRIPTION
**Pull request by edleeks87**
# Description
Fixes the publish issue where `lib` and `node_modules` directories were not present in published package

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "carbon-state-management",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (samhinton88) [inline package-lock.json]: I think this may already be at `0.0.2`, would a change not bump it to `0.0.3`?

Comment (edleeks87) [reply package-lock.json]: Yeah that would be the next step of releasing it surely?
Also it's only on `0.0.1` currently in master

Comment (samhinton88) [reply package-lock.json]: That's odd as it's on `0.0.2` on npm.

Comment (samhinton88) [reply package-lock.json]: If you're cutting a tag from this branch it should have the next version number in the `package.json`?

Comment (edleeks87) [reply package-lock.json]: Just realised there's no release branch so I suppose I can do it all in one haha

**Pull request merged**